internal/ast/symbol.go

```diff
@@ -16,7 +16,7 @@ type Symbol struct {
 	ValueDeclaration             *Node
 	Members                      SymbolTable
 	Exports                      SymbolTable
-	id                           atomic.Uint64
+	Id                           atomic.Uint64
```

Comment:
I don't think we want to make this public until we start preallocating them unconditionally; asking for the ID via `GetNodeId` should be required.